File: 7/Santonacci.java
class Santonacci {
    public int fibonacci(int n) {
        int sum = 0;
        int n1=0;
        int n2=1;
        if(n == 0){
            return 0;
        } else if(n == 1){
            return n1+n2;
        } else {
            for(int j=1;j<n;++j){
                sum= n1+n2;
                n1=n2;
                n2=sum;
            }
        }
        return sum;
    }
}

File: 7/Santonacci.kt
class Santonacci {
    fun fibonacci(n: Int): Int {
        return when(n) {
            0 -> 0
            1 -> 1
            else -> {
                var previousNumberA = 0 // We always start at 0
                var previousNumberB = 1 // But previous one should be at least 1
                var currentFibonacci = 0
                for(index in 2 .. n) {
                    // first sum them up
                    currentFibonacci = previousNumberA + previousNumberB
                    // then replace previousA with previousB
                    previousNumberA = previousNumberB
                    // and previousB becomes current sum
                    previousNumberB = currentFibonacci
                }
                return currentFibonacci
            }
        }
    }
}
